### PR TITLE
[WIP] Pd facilitator program registration

### PR DESCRIPTION
**Pull request by Hamms**

### CODE REVIEW

File: dashboard/db/migrate/20170330214255_create_pd_program_registration.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreatePdProgramRegistration < ActiveRecord::Migration[5.0]
+  def change
+    create_table :pd_facilitator_program_registrations do |t|
+      t.references :user, index: true
+      t.text :form_data
+
+      t.timestamps null: false
+    end
+  end
+end
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170330214255_create_pd_program_registration.rb:10]: Please make the migration its own separate PR.

Comment (Hamms) [reply dashboard/db/migrate/20170330214255_create_pd_program_registration.rb:10]: Why? Given that the model is dependent on this migration, shouldn't they be in the same PR?

Comment (ashercodeorg) [reply dashboard/db/migrate/20170330214255_create_pd_program_registration.rb:10]: In my (possibly overcareful) world, the migration reaches all environments, then the model reaches all environments, and then the usage of the model reaches all environments.

If the migration and model (its usage) are PRed together, then if the migration fails (alternately another (earlier) migration in that deploy), the model refers to a non-existent table.

**Comment by Hamms** [comment]
Closing in favor of https://github.com/code-dot-org/code-dot-org/pull/14298, https://github.com/code-dot-org/code-dot-org/pull/14300, and https://github.com/code-dot-org/code-dot-org/pull/14301
